src/app/page.tsx

```
@@ -1,116 +1,75 @@
-		queryOptions: { enabled: isERC1155Query.data },
-	});
-
-	const claimCondition1155 = useReadContract(getActiveClaimCondition1155, {
-		contract,
-		tokenId,
-		queryOptions: {
-			enabled: isERC1155Query.data,
-		},
-	});
-
-	const claimCondition721 = useReadContract(getActiveClaimCondition721, {
-		contract,
-		queryOptions: { enabled: isERC721Query.data },
-	});
+import { defaultTokenId, contract } from "@/lib/constants";
+// lib imports for fetching NFT details
+import { getERC20Info } from "@/lib/erc20";
+import { getERC721Info } from "@/lib/erc721";
+import { getERC1155Info } from "@/lib/erc1155";
+// thirdweb imports
+import { isERC1155 } from "thirdweb/extensions/erc1155";
+import { isERC721 } from "thirdweb/extensions/erc721";
 
-	const claimCondition20 = useReadContract(getActiveClaimCondition20, {
-		contract,
-		queryOptions: { enabled: !isERC721Query.data && !isERC1155Query.data },
-	});
+async function getERCType() {
+  try {
+    const [isErc721, isErc1155] = await Promise.all([
+      isERC721({ contract }).catch(() => false),
+      isERC1155({ contract }).catch(() => false),
+    ]);
 
-	const displayName = isERC1155Query.data
-		? nftQuery.data?.metadata.name
-		: contractMetadataQuery.data?.name;
-
-	const description = isERC1155Query.data
-		? nftQuery.data?.metadata.description
-		: contractMetadataQuery.data?.description;
-
-	const priceInWei = isERC1155Query.data
-		? claimCondition1155.data?.pricePerToken
-		: isERC721Query.data
-			? claimCondition721.data?.pricePerToken
-			: claimCondition20.data?.pricePerToken;
-
-	const currency = isERC1155Query.data
-		? claimCondition1155.data?.currency
-		: isERC721Query.data
-			? claimCondition721.data?.currency
-			: claimCondition20.data?.currency;
-
-	const currencyContract = getContract({
-		address: currency || "",
-		chain,
-		client,
-	});
+    return isErc1155 ? "ERC1155" : isErc721 ? "ERC721" : "ERC20";
+  } catch (error) {
+    console.error("Error detecting ERC type:", error);
```

You're already catching in the promises, what's left that can throw here? It's either one or the other.

I'm this case since the erc type is so critical, I wouldn't catch in the promises and silently return false.

Let it throw